src/test/java/com/github/dariobalinzo/ElasticSourceConnectorTest.java

```diff
@@ -67,4 +75,65 @@ public void shouldGetTaskFromFixedList() {
         connector.stop();
     }
 
+    @Test
+    public void shouldUpdateListOfTopics() throws IOException {
+        throw new UnsupportedOperationException();
+        /*
+        We need to mock the context but I don't know how to do it.
```

Comment:
You can take a look here:
https://github.com/DarioBalinzo/kafka-connect-elasticsearch-source/blob/master/src/test/java/com/github/dariobalinzo/task/ElasticSourceTaskTest.java

I used mockito to instantiate a mock context. 

    `
    @Mock
    private SourceTaskContext context;

    @Before
    public void init() {
        MockitoAnnotations.initMocks(this);
    }
`

In your case you should not need any other mock initialization, since the requestTaskReconfiguration() is returning void.

Then you may check if the requestTaskReconfiguration() is invoked when you add a new index:
`Mockito.verify(context, atLeast(1)).requestTaskReconfiguration();`

I hope this helps